app/src/main/java/com/example/barterbuddy/activities/BarterPage.java

```diff
@@ -36,16 +72,127 @@ protected void onCreate(Bundle savedInstanceState) {
 
     getXmlElements();
 
-    // listener for abort trade button, closes trade and starts "tinder" activity
+    loadPage();
+  }
+
+  private void loadPage() {
```

Comment:
I'd prefer to leave it in. Because the methods need to call each other to work around the asynchronous Firebase call issue, the methods calling each other acts as a kind of checklist.